climada/hazard/tc_tracks_synth.py

```diff
@@ -99,8 +143,25 @@ def calc_perturbed_trajectories(tracks,
         Random number generator seed for replicability of random walk.
         Put negative value if you don't want to use it. Default: configuration file.
     decay : bool, optional
-        Whether to compute land decay in probabilistic tracks. Default: True.
+        Whether to apply landfall decay in probabilistic tracks. Default: True.
+    use_global_decay_params : bool, optional
+        Whether to use precomputed global parameter values for landfall decay
+        obtained from IBTrACS (1980-2019). If False, parameters are fitted
+        using historical tracks in input parameter 'tracks', in which case the
+        landfall decay applied depends on the tracks passed as an input and may
+        not be robust if few historical tracks make landfall in this object.
+        Default: True.
     """
+    
+    # Remove tracks with a single point
+    single_point_ids = [track.attrs['sid'] for track in tracks.data if track.time.size < 2]
+    if len(single_point_ids) > 0:
+        LOGGER.info('%s storm events are discarded because they consist of a '
+                    'single point: %s.', len(single_point_ids),
+                    ", ".join(single_point_ids))
+        tracks.data = [track for track in tracks.data if track.time.size >= 2]
```

Comment:
Discarding data without reason is not a good idea. Hence, I propose to simply ignore the single point events for the synthetic track generation. This is done here by saving the single point tracks in line 157, and appending them again to the full set in line 233.

```suggestion
    # Do no use tracks with a single time point for synthetic track generation
    single_point_tracks = [track for track in tracks.data if track.time.size <= 1]
    if single_point_tracks:
        LOGGER.info('%s storm events consist of a single time point. No synthetic'
                    'track is generated for these events: %s.', len(single_point_tracks),
                    ", ".join([track.attrs['sid'] for track in single_points_tracks]))
        tracks.data = [track for track in tracks.data if track.time.size >= 2]
```